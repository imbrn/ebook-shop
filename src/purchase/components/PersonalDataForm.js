import React from "react";
import StepPanel, { StepPanelField, TextField } from "./StepPanel";
import { Formik } from "formik";
import nanoid from "nanoid";

export const PersonalDataForm = ({
  number,
  title,
  open,
  personalData,
  updatePersonalData
}) => {
  const handleInputChange = (e, handleChange) => {
    updatePersonalData({
      [e.target.name]: e.target.value
    });
    handleChange(e);
  };

  const nameFieldId = nanoid(8);
  const emailFieldId = nanoid(8);
  const cpfFieldId = nanoid(8);

  return (
    <StepPanel number={number} title={title} open={open}>
      <Formik
        onSubmit={e => e.preventDefault()}
        initialValues={personalData}
        render={({
          values,
          errors,
          touched,
          handleChange,
          handleBlur,
          handleSubmit
        }) => (
          <form onSubmit={handleSubmit}>
            <StepPanelField
              name="name"
              label="Name"
              placeholder="Your name here"
              fieldId={nameFieldId}
              error={touched.name && errors.name}
              component={TextField}
              value={values.name}
              onChange={e => handleInputChange(e, handleChange)}
              onBlur={handleBlur}
            />

            <StepPanelField
              name="email"
              label="E-email"
              placeholder="Your e-mail here"
              fieldId={emailFieldId}
              error={touched.email && errors.email}
              component={TextField}
              value={values.email}
              onChange={e => handleInputChange(e, handleChange)}
              onBlur={handleBlur}
            />

            <StepPanelField
              name="cpf"
              label="CPF"
              placeholder="Your CPF here"
              fieldId={cpfFieldId}
              error={touched.cpf && errors.cpf}
              component={TextField}
              value={values.cpf}
              onChange={e => handleInputChange(e, handleChange)}
              onBlur={handleBlur}
            />
          </form>
        )}
      />
    </StepPanel>
  );
};

PersonalDataForm.defaultProps = {
  number: 1,
  title: "Personal data",
  open: true,
  personalData: {
    name: "",
    email: "",
    cpf: ""
  }
};
