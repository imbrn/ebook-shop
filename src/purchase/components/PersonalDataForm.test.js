import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import { PersonalDataForm } from "./PersonalDataForm";

jest.mock("nanoid", () => {
  let nextId = 0;
  return () => nextId++;
});

afterEach(cleanup);

test("renders correctly", () => {
  const personalData = {
    name: "Customer Name",
    email: "customer@email.com",
    cpf: "12345678910"
  };
  const { container } = render(
    <PersonalDataForm personalData={personalData} />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("updatePersonalData", () => {
  const updatePersonalData = jest.fn();
  const { getByPlaceholderText } = render(
    <PersonalDataForm updatePersonalData={updatePersonalData} />
  );

  fireEvent.change(getByPlaceholderText("Your name here"), {
    target: { value: "My Name" }
  });

  fireEvent.change(getByPlaceholderText("Your e-mail here"), {
    target: { value: "my@email.com" }
  });

  fireEvent.change(getByPlaceholderText("Your CPF here"), {
    target: { value: "12345678910" }
  });

  expect(updatePersonalData.mock.calls).toEqual([
    [{ name: "My Name" }],
    [{ email: "my@email.com" }],
    [{ cpf: "12345678910" }]
  ]);
});
