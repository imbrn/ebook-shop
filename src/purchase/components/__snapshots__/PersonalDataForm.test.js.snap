// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header>
    <span
      class=""
    >
      1
    </span>
    <span
      class=""
    >
      Personal data
    </span>
  </header>
  <div>
    <form>
      <div>
        <div>
          <label
            class=""
            for="0"
          >
            Name
          </label>
        </div>
        <div>
          <input
            class=""
            id="0"
            name="name"
            placeholder="Your name here"
            type="text"
            value="Customer Name"
          />
        </div>
      </div>
      <div>
        <div>
          <label
            class=""
            for="1"
          >
            E-email
          </label>
        </div>
        <div>
          <input
            class=""
            id="1"
            name="email"
            placeholder="Your e-mail here"
            type="text"
            value="customer@email.com"
          />
        </div>
      </div>
      <div>
        <div>
          <label
            class=""
            for="2"
          >
            CPF
          </label>
        </div>
        <div>
          <input
            class=""
            id="2"
            name="cpf"
            placeholder="Your CPF here"
            type="text"
            value="12345678910"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
